terraform/main.tf

```diff
@@ -222,32 +222,39 @@ resource "aws_ses_receipt_rule" "ffis_ingest" {
   }
 }
 
-resource "aws_iam_policy" "ses_source_data_s3_access" {
-  name        = "${var.namespace}-ses_source_data_s3_access"
-  path        = "/"
-  description = "Allows SES to putObject into Grants source data bucket"
+data "aws_iam_policy_document" "ses_source_data_s3_access" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["ses.amazonaws.com"]
+    }
 
-  policy = jsonencode({
-    Version = "2012-10-17"
-    Statement = [
-      {
-        Principal = {
-          Service = "ses.amazonaws.com"
-        }
-        Action = [
-          "s3:PutObject",
-        ]
-        Effect   = "Allow"
-        Resource = "${module.grants_source_data_bucket.bucket_id}/ses/*"
-        Condition = {
-          StringEquals = {
-            "AWS:SourceArn"     = "arn:aws:ses:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:receipt-rule-set/ffis_ingest-rule-set:receipt-rule/ffis_ingest-${var.environment}"
-            "AWS:SourceAccount" = data.aws_caller_identity.current.account_id
-          }
-        }
-      },
+    actions = [
+      "s3:PutObject",
     ]
-  })
+
+    resources = ["${module.grants_source_data_bucket.bucket_id}/ses/*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "AWS:SourceArn"
+      values = [
+        "arn:aws:ses:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:receipt-rule-set/ffis_ingest-rule-set:receipt-rule/ffis_ingest-${var.environment}"
+      ]
```

Comment:
If the answer to the my question regarding the `depends_on` is "no", this can probably get changed back to referencing the `aws_ses_receipt_rule.ffis_ingest` resource directly.

Reply:
See my previous reply, I think directly referencing the receipt rule ARN will still create a cyclical dependency. It may be more obfuscated in this case since I'm using the IAM policy document data source instead of the resource.  And by "obfuscated", I mean that I'm not sure Terraform will even pick on on the cyclical dependency and just try to create rule without making sure the bucket policy is applied.  The problem here is that the receipt rule checks to see if it can write to the bucket as it's being created and if it can't it will throw the error we're seeing.  So, by trying to reference the resource's output, we're implying that the resource was able to create successfully (which it can't because it needs that policy in place to check write access -- and around we go).  Hope that makes sense :-)

Reply:
Got it- thanks for the details!